### DISCUSSION TITLE
Problem when using CGAL Clip tool

### DISCUSSION THREAD

**Discussion by Giseribility** [CGAL/cgal#6949] [Q&A]
When using the PMP Clip tools of CGAL to clip polyhedron with face (.obj format), the program runs successfully, but the clip result is not  topologically correct, there is a single line in the result polyhedron. I wonder what causes the incorrect result,  and how to avoid it?

the polyhedron and the face:
![image](https://user-images.githubusercontent.com/46023666/194860306-c7ff56cc-049c-4c9c-ab60-37284f322ffa.png)

the clip result, 3 facets were deleted to show the single line:
![image](https://user-images.githubusercontent.com/46023666/194859932-cb63c3b8-40a2-4c7c-a2cf-536246721648.png)

the three model data (.obj format) is as follows:
https://gist.github.com/Giseribility/2d0d3a04ac73db77c74a410473eef03c

**Comment by sloriot** [comment]
I don't understand why you are saying that the output is not correct. `face` splits the volume defined by `polyhedron` in 2 sub-volumes. Only the volume on the *negative side* of `face` is kept.

**Reply by Giseribility** [reply]
but the volume on the negative side of face, its facets structure are not correct. Here is a facet degenerated into nearly a polyline (marked as red in the following figure). 
![image](https://user-images.githubusercontent.com/46023666/194976587-98938cf3-f5d5-4c6b-b60e-0d69877ab6aa.png)

Open the obj file as txt, we can see that the coordinate of some vertex is very close, how to avoid the slender triangle when using cilp tool?
![image](https://user-images.githubusercontent.com/46023666/194977187-83fc88b5-eab7-49b6-85be-607a2996c4d9.png)
